### DISCUSSION TITLE
Move with path Component?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1176] [Q&A] [answered]
Hi:

Recently we got a feedback from students that
Is it possible to have a component that moving entity with a path?
If we only want to move view component then we could use path transition
but if we only move view component without moving entity then collision handler will not work during the transition
so is it possible to have a path moving component or move with path component?
And we also do some research and found the algorithm of path interpolation here
https://stackoverflow.com/questions/30555912/move-node-along-path-without-pathtransition
so it would be nice to have a similar implementation of this mechanism
or there is an existing one we havent found yet? 
If so please let us know

Thank you very much

**Comment by AlmasB** [comment, marked as answer]
Hi, not quite the same and not perfect as it interpolates rather than computes, but you can try:

```
animationBuilder().translate(e).alongPath(Path path)
```

Let me know how it goes.

**Reply by chengenzhao** [reply]
it works
thx
